types/mempool/sender_nonce.go

```diff
@@ -65,6 +87,9 @@ func (snm *senderNonceMempool) setSeed(seed int64) {
 // Insert adds a tx to the mempool. It returns an error if the tx does not have at least one signer.
 // priority is ignored.
 func (snm *senderNonceMempool) Insert(_ sdk.Context, tx sdk.Tx) error {
+	if snm.maxTx > 0 && len(snm.existingTx) >= snm.maxTx {
```

Comment:
Does this condition fulfill on the generating [comment](https://github.com/cosmos/cosmos-sdk/pull/13888#pullrequestreview-1189945991) for the the generating issue #13974 for this PR? 

> left one question on mempool size as not all node operators need to run this mempool, so it could be they "turn it off" by setting the max allowed txs to 0.

if `maxTx` is 0 we add txs to the mempool indefinitely, which isn't opting out of an app-side mempool as described above.

Reply:
Yes, there is a comment thread above addressing this. if you want we can continue the conversation in that thread https://github.com/cosmos/cosmos-sdk/pull/14014#issuecomment-1328414148